src/vos/vos_tree.c

```diff
@@ -690,7 +690,12 @@ svt_rec_update(struct btr_instance *tins, struct btr_record *rec,
 	if (rc != 0)
 		return rc;
 
-	return svt_rec_alloc_common(tins, rec, skey, rbund);
+	rc = svt_rec_alloc_common(tins, rec, skey, rbund);
+	if (rc != 0)
+		return rc;
+
+	/* Inform btr_update() that the original record is replaced */
+	return 1;
```

Comment:
[Stupid question] So `svt_rec_update` will never return 0? Can't `btr_update` just handle `rc == 0` case (suppose that we don't make any change to `svt_rec_update`)?

Reply:
For svt_rec_update() (registered for SV tree tyoe), it only returns -DER_NO_PERM or 1, but other tree type may register ".to_rec_update()" returning 0. (for in-place overwrite or just skip the update operation).

Reply:
Ah, indeed. I would consider updating the API doc of `to_rec_update` to highlight this new return value.

Reply:
missed this reply before pushing, I'll update the comment if the patch needs be refreshed (it's highly likely). thanks!